### DISCUSSION TITLE
Inline Event Handler in OIDC Form Post Response

### DISCUSSION THREAD

**Discussion by jamesshuriff** [keycloak/keycloak#32054] [Ideas]
The OIDC Form Post Response has an inline script that's used to submit the form. This makes it impossible to define a strong Content Security Policy. I would like to propose moving the inline script into a dedicated endpoint to eliminate the necessity for "unsafe-inline" in Content Security Policies. I'm preparing an issue and PR to implement the change but would like to get some feedback, here.

**Comment by jamesshuriff** [comment]
One of the contributors shot down my PR because they did not see the value of the change. Obviously, I disagree, or I would not have written a PR. 🤷

**Reply by jonkoops** [reply]
We appreciate your willingness to contribute, but the comment was simply that the technical implementation is not a fix for the larger issue, which I already explained under https://github.com/keycloak/keycloak/pull/32057#issuecomment-2284341128.
